bc/channel/management/commands/post.py

```diff
@@ -1,49 +1,95 @@
+from rq import Retry
 
-from bc.channel.models import Channel
-from bc.channel.utils.connectors.masto import MastodonConnector
-from bc.channel.utils.connectors.twitter import TwitterConnector
+from bc.channel.models import Channel, Group
+from bc.core.utils.commands import (
+    show_all_channels_table,
+    show_channel_groups_table,
+)
+
+queue = get_queue("default")
+
+
+def handle_post_command(
+    ids: list[int], mapping: dict[int, Channel | Group], text: str
+) -> None:
+    """
+    Enqueues a job to create a post for each channel in the list of ids provided
+    by the user.
+
+    Args:
+        ids (list[int]): list of the id from the user's input.
+        mapping (dict[int, Channel  |  Group]): Mapping of objects in the table.
+        text (str): Text to include in the posts.
+
+    Raises:
+        ValueError: if the provided input is not in the mapping variable.
+    """
+    for record_id in ids:
+        record = mapping.get(record_id)
+
+        if record is None:
+            raise ValueError(f"No channel {record_id}")
+
+        if isinstance(record, Group):
+            for channel in record.channels.all():
+                api = channel.get_api_wrapper()
+                queue.enqueue(
+                    api.add_status,
+                    text,
+                    None,
+                    retry=Retry(
+                        max=settings.RQ_MAX_NUMBER_OF_RETRIES,
+                        interval=settings.RQ_RETRY_INTERVAL,
+                    ),
+                )
+        else:
+            api = record.get_api_wrapper()
+            queue.enqueue(
+                api.add_status,
+                text,
+                None,
+                retry=Retry(
+                    max=settings.RQ_MAX_NUMBER_OF_RETRIES,
+                    interval=settings.RQ_RETRY_INTERVAL,
+                ),
+            )
 
 
 class Command(BaseCommand):
     help = "Post something manually to one or more channels."
 
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--show_groups",
+            action="store_true",
+            help="Shows the list of groups instead of individual channels",
+        )
+
     def handle(self, *args, **options):
-        tab = PrettyTable()
-        channel_mapping = {}
-        tab.field_names = ["ID", "Service", "Account", "Enabled", "URL"]
-        db_channels = Channel.objects.all()
-        for channel in db_channels:
-            tab.add_row(
-                [
-                    channel.id,
-                    channel.service,
-                    channel.account,
-                    channel.enabled,
-                    channel.self_url(),
-                ]
-            )
-            channel_mapping[channel.id] = channel
-        self.stdout.write(self.style.SUCCESS(tab))
+        if not options["show_groups"]:
+            table, mapping = show_all_channels_table()
+        else:
+            table, mapping = show_channel_groups_table()
+
```

Comment:
I think the logic here should be flipped so the default is a grouped table and you have to use a special arg to get the *un*grouped results.

Reply:
thanks for your comment. I'll change the logic here.